src/tools/translate-to-fuzz.h

```diff
@@ -0,0 +1,1106 @@
+    wasm.memory.exists = true;
+    // use one page
+    wasm.memory.initial = wasm.memory.max = 1;
+  }
+
+  const Name HANG_LIMIT_GLOBAL = "hangLimit";
+
+  void addHangLimitSupport() {
+    auto* glob = new Global;
+    glob->name = HANG_LIMIT_GLOBAL;
+    glob->type = i32;
+    glob->init = builder.makeConst(Literal(int32_t(HANG_LIMIT)));
+    glob->mutable_ = true;
+    wasm.addGlobal(glob);
+
+    auto* func = new Function;
+    func->name = "hangLimitInitializer";
+    func->result = none;
+    func->body = builder.makeSetGlobal(glob->name,
+      builder.makeConst(Literal(int32_t(HANG_LIMIT)))
+    );
+    wasm.addFunction(func);
+
+    auto* export_ = new Export;
+    export_->name = func->name;
+    export_->value = func->name;
+    export_->kind = ExternalKind::Function;
+    wasm.addExport(export_);
+  }
+
+  Expression* makeHangLimitCheck() {
+    return builder.makeSequence(
+      builder.makeIf(
+        builder.makeUnary(
+          UnaryOp::EqZInt32,
+          builder.makeGetGlobal(HANG_LIMIT_GLOBAL, i32)
+        ),
+        makeTrivial(unreachable)
+      ),
+      builder.makeSetGlobal(
+        HANG_LIMIT_GLOBAL,
+        builder.makeBinary(
+          BinaryOp::SubInt32,
+          builder.makeGetGlobal(HANG_LIMIT_GLOBAL, i32),
+          builder.makeConst(Literal(int32_t(1)))
+        )
+      )
+    );
+  }
+
+  void addDeNaNSupport() {
+    auto add = [&](Name name, WasmType type, Literal literal, BinaryOp op) {
+      auto* func = new Function;
+      func->name = name;
+      func->params.push_back(type);
+      func->result = type;
+      func->body = builder.makeIf(
+        builder.makeBinary(
+          op,
+          builder.makeGetLocal(0, type),
+          builder.makeGetLocal(0, type)
+        ),
+        builder.makeGetLocal(0, type),
+        builder.makeConst(literal)
+      );
+      wasm.addFunction(func);
+    };
+    add("deNaN32", f32, Literal(float(0)), EqFloat32);
+    add("deNaN64", f64, Literal(double(0)), EqFloat64);
+  }
+
+  Expression* makeDeNaNOp(Expression* expr) {
+    if (!DE_NAN) return expr;
+    if (expr->type == f32) {
+      return builder.makeCall("deNaN32", { expr }, f32);
+    } else if (expr->type == f64) {
+      return builder.makeCall("deNaN64", { expr }, f64);
+    }
+    return expr; // unreachable etc. is fine
+  }
+
+  // function generation state
+
+  Function* func;
+  std::vector<Expression*> breakableStack; // things we can break to
+  Index labelIndex;
+
+  // a list of things relevant to computing the odds of an infinite loop,
+  // which we try to minimize the risk of
+  std::vector<Expression*> hangStack;
+
+  std::map<WasmType, std::vector<Index>> typeLocals; // type => list of locals with that type
+
+  void addFunction() {
+    Index num = wasm.functions.size();
+    func = new Function;
+    func->name = std::string("func_") + std::to_string(num);
+    func->result = getReachableType();
+    assert(typeLocals.empty());
+    Index numParams = upTo(upTo(5));
```

Comment:
`upTo(upTo(x))` is non-obvious at first glance that it's skewing the distribution. (Made further non-obvious by the name `upTo` that hides that the number is even random)

How about wrapping this idiom in a function:

```c++
Index getLessThanWeightedLow(Index limit) {
  return getLessThan(getLessThan(limit));
}
```